casr-afl is unaware of out/default/crashes.DATE directories

when using casr-afl on an output directory that has restarts in it, it does not find saved crashes directories.
```
 $ ls out3/default/
cmdline                      fuzz_bitmap   hangs.2023-05-11-08:13:06
crashes                      fuzzer_setup  hangs.2023-06-07-10:04:14
crashes.2023-05-11-08:13:06  fuzzer_stats  plot_data
```
casr-afl needs not only to browse `crashes/` but check all `crashes*/` directories.

> btw is there a workaround for this: `[ERROR] Error: Out of memory: hard_rss_limit_mb exhausted for input: ...` ?

`casr-san` [sets](https://github.com/ispras/casr/blob/master/casr/src/bin/casr-san.rs#L107) hard_rss_limit_mb to 2048 by default. You can override this by setting `ASAN_OPTIONS=hard_rss_limit_mb=0`.

--- a/casr/src/bin/casr-afl.rs
+++ b/casr/src/bin/casr-afl.rs
@@ -213,19 +213,18 @@ fn main() -> Result<()> {
         }
 
         // Push crash paths.
-        for crash in fs::read_dir(path.join("crashes"))? {
-            let crash_path = crash?.path();
-            let fname = crash_path
-                .file_name()
-                .unwrap()
-                .to_str()
-                .unwrap()
-                .to_string();
-
-            if fname.starts_with("id:") {
-                crash_info.path = crash_path.to_path_buf();
-                crashes.insert(fname, crash_info.clone());
-            }
+        for crash in path
+            .read_dir()?
+            .flatten()
+            .filter(|e| e.file_name().into_string().unwrap().starts_with("crashes"))
+            .flat_map(|e| e.path().read_dir())
+            .flatten()
+            .flatten()
+            .filter(|e| e.file_name().into_string().unwrap().starts_with("id:"))
+        {
+            let mut info = crash_info.clone();
+            info.path = crash.path();
+            crashes.insert(crash.file_name().into_string().unwrap(), info);
         }
     }
 

